README.md

```diff
@@ -144,6 +144,33 @@ so you can simply import the NIST level you want to play with:
 The above example would also work with the other NIST levels
 `ML_DSA_65` and `ML_DSA_87`.
 
+#### Hash ML-DSA
+
+Following algorithms 4 and 5 of FIPS 204 we also include a version of pre-hash ML-DSA which hashes the message before signing it using SHA512 by default for
+all three security levels. This is used in much the same way as ML-DSA:
+
+```python
+>>> from dilithium_py.ml_dsa import HASH_ML_DSA_44_WITH_SHA512
+>>>
+>>> # Example of signing
+>>> pk, sk = HASH_ML_DSA_44_WITH_SHA512.keygen()
+>>> msg = b"Your message signed by ML_DSA"
+>>> sig = HASH_ML_DSA_44_WITH_SHA512.sign(sk, msg)
```

Comment:
wait... but the whole point of HashML-DSA is so that the message can be externally hashed, and then it's the hash that's signed, not the message directly...